src/Microsoft.Azure.SignalR/ServiceRouteBuilder.cs

```diff
@@ -58,6 +59,17 @@ public ServiceRouteBuilder(RouteBuilder routes)
 
             var dispatcher = _serviceProvider.GetRequiredService<ServiceHubDispatcher<THub>>();
             dispatcher.Start(app);
+
+#if NETCOREAPP
+            var lifetime = _serviceProvider.GetService<IHostApplicationLifetime>();
+#elif NETSTANDARD
+            var lifetime = _serviceProvider.GetService<IApplicationLifetime>();
+#endif

```

Comment:
>endif [](start = 1, length = 5)

otherwise throw?

Reply:
simply disable the feature could be better?